src/test/java/com/bravo/user/service/PaymentServiceTest.java

```diff
@@ -0,0 +1,141 @@
+    public void retrievePaymentByUserId() {
+        final String userId = "1";
+        final PageRequest pageRequest = PageUtil.createPageRequest();
+        final List<PaymentDto> results = paymentService.retrievePaymentByUserId(userId, pageRequest, httpResponse);
+        assertEquals(paymentDtos, results);
+        assertEquals("6", httpResponse.getHeader(PAGE_COUNT));
+        assertEquals("1", httpResponse.getHeader(PAGE_NUMBER));
+        assertEquals("20", httpResponse.getHeader(PAGE_SIZE));
+
+        final PaymentFilter filter = new PaymentFilter(userId);
+        final PaymentSpecification specification = new PaymentSpecification(filter);
+        verify(paymentRepository).findAll(specification, pageRequest);
+    }
+
+    @Test
+    public void retrievePaymentByUserIdWithPagination() {
+        final String userId = "1";
+        final PageRequest pageRequest = PageUtil.createPageRequest(1, 6);
+        final List<PaymentDto> results = paymentService.retrievePaymentByUserId(userId, pageRequest, httpResponse);
+        assertEquals(paymentDtos, results);
+        assertEquals("6", httpResponse.getHeader(PAGE_COUNT));
+        assertEquals("1", httpResponse.getHeader(PAGE_NUMBER));
+        assertEquals("6", httpResponse.getHeader(PAGE_SIZE));
+
+        final PaymentFilter filter = new PaymentFilter(userId);
+        final PaymentSpecification specification = new PaymentSpecification(filter);
+        verify(paymentRepository).findAll(specification, pageRequest);
+    }
+
+    @Test
+    public void retrieveWithFilter() {
+        final PageRequest pageRequest = PageUtil.createPageRequest();
+        final List<PaymentDto> results = paymentService.retrieve(paymentFilter, pageRequest, httpResponse);
+        assertEquals(paymentDtos, results);
+        assertEquals("6", httpResponse.getHeader(PAGE_COUNT));
+        assertEquals("1", httpResponse.getHeader(PAGE_NUMBER));
+        assertEquals("20", httpResponse.getHeader(PAGE_SIZE));
+
+        final PaymentSpecification specification = new PaymentSpecification(paymentFilter);
+        verify(paymentRepository).findAll(specification, pageRequest);
+
+    }
+
+    @Test
+    public void retrieveWithFilterWithPagination() {
+        final PageRequest pageRequest = PageUtil.createPageRequest(1, 6);
+        final List<PaymentDto> results = paymentService.retrieve(paymentFilter, pageRequest, httpResponse);
+        assertEquals(paymentDtos, results);
+        assertEquals("6", httpResponse.getHeader(PAGE_COUNT));
+        assertEquals("1", httpResponse.getHeader(PAGE_NUMBER));
+        assertEquals("6", httpResponse.getHeader(PAGE_SIZE));
+
+        final PaymentSpecification specification = new PaymentSpecification(paymentFilter);
+        verify(paymentRepository).findAll(specification, pageRequest);
+
+    }
+
+    private Payment createPayment(String paymentId)
+    {
+        final Payment payment = new Payment();
+        payment.setId(paymentId);
+        return payment;
+    }
```

Comment:
In my view, this is too much work to make a payment object.  The constructor or builder can be better alternatives.

Reply:
Agreed. This is an oversight on my part. I will look into improving this.

Reply:
The @Builder seems to be a good option for here, rather than what I did before with creating a single arg constructor, it's less messy than potentially having to add more constructors if I need to just set specific values.